py/tests/unit/with_runtime_sparkling/test_glm.py

```diff
@@ -60,3 +61,29 @@ def testPropagationOfPredictionCol(prostateDataset):
     model = algo.fit(prostateDataset)
     columns = model.transform(prostateDataset).columns
     assert True == (predictionCol in columns)
+
+
+def testInteractionPairsAffectResult(airlinesDataset):
+    [traningDataset, testingDataset] = airlinesDataset.randomSplit([0.95, 0.05], 1)
+    def createInitialGlmDefinition():
+        return H2OGLM(
+            seed=42,
+            family="binomial",
+            lambdaSearch=True,
+            featuresCols=["Year", "Month", "DayofMonth", "DayOfWeek", "CRSDepTime", "CRSArrTime", "UniqueCarrier",
+                         "CRSElapsedTime", "Origin", "Dest", "Distance"],
+            labelCol="IsDepDelayed")
+
+    referenceDeepLearning = createInitialGlmDefinition()
+    referenceModel = referenceDeepLearning.fit(traningDataset)
+    referenceResult = referenceModel.transform(testingDataset)
+
+    glm = createInitialGlmDefinition()
+    interactionPairs = [("CRSDepTime", "UniqueCarrier"),
+                        ("CRSDepTime", "Origin"),
+                        ("UniqueCarrier", "Origin")]
+    glm.setInteractionPairs(interactionPairs)
+    model = glm.fit(traningDataset)
+    result = model.transform(testingDataset)
```

Comment:
@michalkurka @wendycwong I have another problem with `GlmMojoModel`. If I train mojo model with interaction pairs, I get the following exception:
```
 java.lang.ArrayIndexOutOfBoundsException: 14
                at hex.genmodel.algos.glm.GlmMojoModel.glmScore0(GlmMojoModel.java:50)
                at hex.genmodel.algos.glm.GlmMojoModelBase.score0(GlmMojoModelBase.java:32)
                at hex.genmodel.easy.EasyPredictModelWrapper.predict(EasyPredictModelWrapper.java:889)
                at hex.genmodel.easy.EasyPredictModelWrapper.preamble(EasyPredictModelWrapper.java:867)
                at hex.genmodel.easy.EasyPredictModelWrapper.predictBinomial(EasyPredictModelWrapper.java:569)
                at ai.h2o.sparkling.ml.models.H2OMOJOPredictionBinomial$$anonfun$getBinomialPredictionUDF$6.apply(H2OMOJOPredictionBinomial.scala:79)
                at ai.h2o.sparkling.ml.models.H2OMOJOPredictionBinomial$$anonfun$getBinomialPredictionUDF$6.apply(H2OMOJOPredictionBinomial.scala:76)
```
Sparkling Water passes the below list of features to `EasyPredictModelWrapper` regardless what parameters were used for training:
```"Year", "Month", "DayofMonth", "DayOfWeek", "CRSDepTime", "CRSArrTime", "UniqueCarrier","CRSElapsedTime", "Origin", "Dest", "Distance"```

Please can you advise what I'm doing wrong? Do I have to combine "interaction_pairs" with another parameter?

Reply:
@mn-mikke Let me chat with @michalk and get back to you.

Reply:
@mn-mikke Interaction is not currently supported in Mojo and I agree that it should be there.  I am writing a JIRA to make sure it will be there.

Reply:
I noticed a similar issue when I was writing another test yesterday.

I don't think the MOJO/POJO actually can handle interactions.

@wendycwong what I think we should do:

1. disable downloading MOJO/POJO for model trained with interactions; document that we currently don't support that

2. add support for interactions in MOJO

this assumes (2) cannot be done quickly, if (2) is actually easy to do we can skip (1)

Reply:
@michalk:  I am not sure that adding the interaction is trivial.  I know it is do-able.

Reply:
@mn-mikke Here is the JIRA to resolve 1: https://0xdata.atlassian.net/browse/PUBDEV-7635?filter=-2

Reply:
Thanks a lot @wendycwong!

Reply:
@michalkurka @wendycwong Is there any Jira ticket for 2:
> 2. add support for interactions in MOJO

Reply:
Jira ticket for the second phase: https://0xdata.atlassian.net/browse/PUBDEV-7776